### ISSUE TITLE
Rework config package

### ISSUE THREAD

**Issue by eddycharly** [jenkins-x/lighthouse#965]
How about splitting the config package per concern ?

I find it a bit hard to follow, things are not really organised and most of the structs are mixed together. 

One concern that could be placed in a separate package could be everything related to jobs ?

**Comment by eddycharly** [comment]
Job config concern was addressed in #968 .

Next is Lighthouse config concern.

**Comment by eddycharly** [comment]
One last thing, shall we move the config package into apis ?
It looks to me that the config is part of the api.

**Comment by abayer** [comment]
I'm debating that - for now, I'd say no...

**Comment by eddycharly** [comment]
Ok, can close the issue.
